reactioncommerce/reaction
# fix: Product prices showing as $NaN.undefined on the customer product grid

Resolves #4467  
Impact: **minor**  
Type: **bugfix**

Closing #4484 in favor of this PR.

## Issue
When a Product is made visible and published without any Variant,  Catalog customer gets price range as **$NaN.undefined - $-NaN.undefined**

## Solution
Add a check in the product catalog publication that makes sure a product has visible `variants` on it before allowing it to be published to the Catalog.


## Testing
- Start the app
- Login as an admin
- Create a product, and make the top level product visible, but DO NOT make the variant visible
- As an admin, see this product is visible to you on the grid (necessary for editing purposes)
- Log out, and see as a customer, this product is not visible to you
